lib/sourcemapper.js

```diff
@@ -0,0 +1,253 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+/** @const */ var _ = require('lodash');
+/** @const */ var async = require('async');
+/** @const */ var fs = require('fs');
+/** @const */ var path = require('path');
+
+/** @const */ var sourceMap = require('source-map');
+
+/** @define {string} */ var MAP_EXT = '.map';
+
+/**
+ * @param {Array.<string>} sourcemapPaths An array of paths to .map sourcemap
+ *  files that should be processed.  The paths should be relative to the 
+ *  current process's current working directory
+ * @param {Logger} logger A logger that reports errors that occurred while
+ *  processing the given sourcemap files
+ * @constructor
+ */
+function Sourcemapper(sourcemapPaths, logger) {
```

Comment:
I concur. Getting rid of the `waitingCallQueue_` would be a nice simplification.